src/s3_fetchers.rs

```diff
@@ -1,6 +1,49 @@
+use async_trait::async_trait;
 use std::str::FromStr;
 
-use aws_sdk_s3::Client;
+use aws_sdk_s3::{
+    output::{get_object_output, GetObjectOutput},
+    Client,
+};
+
+#[async_trait]
+pub trait GetObject {
+    async fn get_object(
+        &self,
+        bucket: &str,
+        prefix: &str,
+    ) -> Result<GetObjectOutput, aws_sdk_s3::types::SdkError<aws_sdk_s3::error::GetObjectError>>;
+}
+
+#[derive(Clone, Debug)]
+pub struct S3GetObject {
+    s3: aws_sdk_s3::Client,
+}
+
+impl S3GetObject {
+    pub fn new(s3: aws_sdk_s3::Client) -> Self {
+        Self { s3 }
+    }
+}
+
+#[async_trait]
+impl GetObject for S3GetObject {
+    async fn get_object(
+        &self,
+        bucket: &str,
+        prefix: &str,
+    ) -> Result<GetObjectOutput, aws_sdk_s3::types::SdkError<aws_sdk_s3::error::GetObjectError>>
+    {
+        Ok(self
+            .s3
+            .get_object()
+            .bucket(bucket)
+            .key(prefix)
```

Comment:
I want to draw your attention that this is not expected to work properly as it is implemented now.

When you do the call to this method (below)

```
.get_object(s3_bucket_name, &format!("{:0>12}", block_height))
```

and here's the code you've replaced 

```
.key(format!("{:0>12}/block.json", block_height))
```

Please note that particular "file name" `block.json`. You're missing it right now. Perhaps it's part of the plan I don't get or a mistake.

Reply:
Woops, thanks for raising, I didn't even notice I took it out :)